services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,869 @@
+                    assert isl.availableBandwidth - protectedIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        when: "Break ISL on the protected path (bring port down) to init the recalculate procedure"
+        def islToBreakProtectedPath = protectedIsls[0]
+        northbound.portDown(islToBreakProtectedPath.dstSwitch.dpId, islToBreakProtectedPath.dstPort)
+
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreakProtectedPath).get().state == IslChangeType.FAILED
+        }
+
+        then: "Protected path is recalculated"
+        def newProtectedPath
+        Wrappers.wait(WAIT_OFFSET) {
+            newProtectedPath = pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath)
+            assert newProtectedPath != currentProtectedPath
+            assert northbound.getFlowStatus(flow.id).status == FlowState.UP
+        }
+
+        and: "Current path is not changed"
+        currentPath == pathHelper.convert(northbound.getFlowPath(flow.id))
+
+        and: "Bandwidth is reserved for new protected path on involved ISLs"
+        def newProtectedIsls = pathHelper.getInvolvedIsls(newProtectedPath)
+        def allLinks = northbound.getAllLinks()
+        def newProtectedIslsInfo = newProtectedIsls.collect { islUtils.getIslInfo(allLinks, it).get() }
+
+        allIsls.each { isl ->
+            newProtectedIslsInfo.each { protectedIsl ->
+                if (isl.id == protectedIsl.id) {
+                    assert isl.availableBandwidth - protectedIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        and: "Reservation is deleted on the broken ISL"
+        def originInfoBrokenIsl = allIsls.find {
+            it.destination.switchId == islToBreakProtectedPath.dstSwitch.dpId &&
+                    it.destination.portNo == islToBreakProtectedPath.dstPort
+        }
+        def currentInfoBrokenIsl = islUtils.getIslInfo(allLinks, islToBreakProtectedPath).get()
+
+        originInfoBrokenIsl.availableBandwidth == currentInfoBrokenIsl.availableBandwidth
+
+        when: "Restore port status"
+        northbound.portUp(islToBreakProtectedPath.dstSwitch.dpId, islToBreakProtectedPath.dstPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreakProtectedPath).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Path is not recalculated again"
+        pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath) == newProtectedPath
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+    }
+
+    @Unroll
+    def "Unable to create #flowDescription flow with protected path if all alternative paths are unavailable"() {
+        given: "Two active neighboring switches with two not overlapping paths at least"
```

Comment:
This requirement sounds strange for me, given the test is aiming for 0 alternative paths.

Reply:
fixed: `"given: "Two active neighboring switches without alt paths""`